packages/frontend-app-api/src/graph/resolveAppNodeSpecs.ts

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  BackstagePlugin,
+  Extension,
+  ExtensionOverrides,
+} from '@backstage/frontend-plugin-api';
+// eslint-disable-next-line @backstage/no-relative-monorepo-imports
+import { toInternalExtensionOverrides } from '../../../frontend-plugin-api/src/wiring/createExtensionOverrides';
+import { ExtensionParameters } from './readAppExtensionsConfig';
+import { AppNodeSpec } from './types';
+
+/** @internal */
+export function resolveAppNodeSpecs(options: {
+  features: (BackstagePlugin | ExtensionOverrides)[];
+  builtinExtensions: Extension<unknown>[];
+  parameters: Array<ExtensionParameters>;
+  forbidden?: Set<string>;
+}): AppNodeSpec[] {
+  const { builtinExtensions, parameters, forbidden = new Set() } = options;
+
+  const plugins = options.features.filter(
+    (f): f is BackstagePlugin => f.$$type === '@backstage/BackstagePlugin',
+  );
+  const overrides = options.features.filter(
+    (f): f is ExtensionOverrides =>
+      f.$$type === '@backstage/ExtensionOverrides',
+  );
+
+  const pluginExtensions = plugins.flatMap(source => {
+    return source.extensions.map(extension => ({ ...extension, source }));
+  });
+  const overrideExtensions = overrides.flatMap(
+    override => toInternalExtensionOverrides(override).extensions,
+  );
+
+  // Prevent core override
+  if (pluginExtensions.some(({ id }) => forbidden.has(id))) {
+    const pluginsStr = pluginExtensions
+      .filter(({ id }) => forbidden.has(id))
+      .map(({ source }) => `'${source.id}'`)
+      .join(', ');
+    const forbiddenStr = [...forbidden].map(id => `'${id}'`).join(', ');
+    throw new Error(
+      `It is forbidden to override the following extension(s): ${forbiddenStr}, which is done by the following plugin(s): ${pluginsStr}`,
+    );
+  }
```

Comment:
It's mirroring the existing behavior, except that this has been refactored to make it possible to list multiple extensions (or none). As for why specifically `'core'` is forbidden to override it's because it's the root of the app with fixed logic that can't be configured. There's really no reason to override it 😅. We added it because we mistakenly overrode it earlier and were confused about why the app broke.